src/angular/components/sso.component.ts

```diff
@@ -66,34 +77,42 @@ export class SsoComponent {
         });
     }
 
-    async submit() {
-        const passwordOptions: any = {
-            type: 'password',
-            length: 64,
-            uppercase: true,
-            lowercase: true,
-            numbers: true,
-            special: false,
-        };
+    async buildAuthorizeUrl(): Promise<string> {
         let codeChallenge = this.codeChallenge;
         let state = this.state;
         if (codeChallenge == null) {
-            const codeVerifier = await this.passwordGenerationService.generatePassword(passwordOptions);
+            const codeVerifier = await this.passwordGenerationService.generatePassword(this.passwordOptions);
             const codeVerifierHash = await this.cryptoFunctionService.hash(codeVerifier, 'sha256');
             codeChallenge = Utils.fromBufferToUrlB64(codeVerifierHash);
             await this.storageService.save(ConstantsService.ssoCodeVerifierKey, codeVerifier);
         }
+
         if (state == null) {
-            state = await this.passwordGenerationService.generatePassword(passwordOptions);
+            state = await this.passwordGenerationService.generatePassword(this.passwordOptions);
+            if (this.returnUri) {
```

Comment:
Also thinking that this should just be a function parameter too.

Reply:
This one didn't have to even be a function parameter. It's constant and scoped just to builAuthorizeUri, so I just made it a variable there.

Reply:
I was referring to redirectUri